src/identity.rs

```diff
@@ -134,6 +134,33 @@ struct TokenAuth<'a> {
     id: &'a str,
 }
 
+// We now check if the string is an Id or Name
+fn is_id(s: &str) -> bool {
+    // example id: 213a290c-ae17-4841-9609-badc01b4ea5c
+    // numbercount:012345678901234567890123456789012345
+    let lens = vec![8, 4, 4, 4, 12];
+    if s.chars().count() != 36 {
+        return false;
+    }
+    s.split('-')
+        .zip(lens)
+        .all(|(s, len)| s.chars().count() == len && s.chars().all(|c| c.is_ascii_hexdigit()))
+}
+
+impl<T> From<T> for IdOrName
+where
+    T: Into<String>,
+{
+    fn from(s: T) -> Self {
+        let unknown = s.into();
+        if is_id(&unknown) {
```

Comment:
I suggest we don't do any guessing and just assume something (we cannot be sure that all IDs are UUIDs, at least keystone domain ID is not). I'd probably go with names.